src/Text/Pandoc/Readers/JATS.hs

```diff
@@ -368,9 +372,13 @@ parseBlock (Elem e) = do
          parseElement = filterChildren isEntry
          wrapWithHeader n mBlocks = do
                       isBook <- gets jatsBook
-                      let n' = if isBook || n == 0 then n + 1 else n
+                      case (filterChild (named “title”) e >>= attrValue “display-as”) of
+	                      Just l -> let n’ = decimal l
+	                      Nothing -> let n' = if isBook || n == 0 then n + 1 else n
```

Comment:
@display-as is a book-specific optional attribute of <title> that dictates at which level the heading needs to be displayed. This code is meant to read (and potentially parse) any given alphanumeric level value, e.g. "level-4", and use that instead of n=0.